ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/ImageSingleVulnerabilities.tsx

```diff
@@ -1,13 +1,208 @@
+    imageVulnerabilities.forEach(({ severity }) => {
+        severityCounts[severity] += 1;
+    });
+
+    return severityCounts;
+}
+
+function statusCountsFromImageVulnerabilities(
+    imageVulnerabilities: ImageVulnerabilitiesResponse['image']['imageVulnerabilities']
+): Record<FixableStatus, number> {
+    const statusCounts = {
+        Fixable: 0,
+        'Not fixable': 0,
+    };
+
+    imageVulnerabilities.forEach(({ isFixable }) => {
+        if (isFixable) {
+            statusCounts.Fixable += 1;
+        } else {
+            statusCounts['Not fixable'] += 1;
+        }
+    });
+
+    return statusCounts;
+}
+
+export type ImageVulnerabilitiesVariables = {
+    id: string;
+};
+
+export type ImageVulnerabilitiesResponse = {
+    image: {
+        imageVulnerabilities: {
+            severity: string;
+            isFixable: boolean;
+        }[];
+    };
+};
+
+export const imageVulnerabilitiesQuery = gql`
+    query getImageVulnerabilities($id: ID!) {
+        image(id: $id) {
+            id
+            imageVulnerabilities {
+                severity
+                isFixable
+            }
+        }
+    }
+`;
+
+export type ImageSingleVulnerabilitiesProps = {
+    imageId: string;
+};
+
+function ImageSingleVulnerabilities({ imageId }: ImageSingleVulnerabilitiesProps) {
+    // TODO Needs integration with URL search filter
+    const { data, loading, error } = useQuery<
+        ImageVulnerabilitiesResponse,
+        ImageVulnerabilitiesVariables
+    >(imageVulnerabilitiesQuery, {
+        variables: { id: imageId },
+    });
+
+    const [activeTabKey, setActiveTabKey] = useCveStatusTabParameter();
+
+    const handleTabClick: TabsProps['onSelect'] = (e, tabKey) => {
+        if (isValidCveStatusTab(tabKey)) {
+            setActiveTabKey(tabKey);
+        }
+    };
+
+    let mainContent: ReactNode | null = null;
+
+    if (error) {
+        mainContent = (
+            <Bullseye>
+                <EmptyState variant={EmptyStateVariant.large}>
+                    <EmptyStateIcon
+                        className="pf-u-danger-color-100"
+                        icon={ExclamationCircleIcon}
+                    />
+                    <Title headingLevel="h2">{getAxiosErrorMessage(error)}</Title>
+                    <EmptyStateBody>Adjust your filters and try again</EmptyStateBody>
+                </EmptyState>
+            </Bullseye>
+        );
+    } else if (loading && !data) {
+        mainContent = (
+            <Bullseye>
+                <Spinner isSVG />
+            </Bullseye>
+        );
+    } else if (data) {
+        const vulnerabilities = data.image.imageVulnerabilities;
+        const severityCounts = severityCountsFromImageVulnerabilities(vulnerabilities);
+        const cveStatusCounts = statusCountsFromImageVulnerabilities(vulnerabilities);
+        // TODO Integrate these with page search filters
+        const hiddenSeverities = new Set<VulnerabilitySeverity>(['LOW_VULNERABILITY_SEVERITY']);
```

Comment:
oh why are we hiding the low severities? is there a possibility of adding new severities to hide in the widget?

Reply:
Oops, no this was just for testing and I forgot to delete it. The hidden severities/statuses are going to be calculated based on search filters once the toolbar is merged. I'll fix this here.